IBPSA/Fluid/SolarCollectors/Controls/CollectorPump.mo

```diff
@@ -0,0 +1,162 @@
+within IBPSA.Fluid.SolarCollectors.Controls;
+model CollectorPump
+  "Controller which activates a circulation pump when solar radiation is above a critical level"
+  extends Modelica.Blocks.Icons.Block;
+
+  parameter Modelica.Units.SI.HeatFlowRate delQ_flow(min=1)=10
+    "Required estimated heat gain per unit area of collector to switch system on";
+  parameter Modelica.Units.SI.Angle azi(displayUnit="deg")
+    "Surface azimuth (0 for south-facing; -90 degree for east-facing; +90 degree for west facing";
+  parameter Modelica.Units.SI.Angle til(displayUnit="deg")
+    "Surface tilt (0 for horizontally mounted collector)";
+  parameter Real rho(
+    final min=0,
+    final max=1,
+    final unit = "1") = 0.2 "Ground reflectance";
+
+  parameter IBPSA.Fluid.SolarCollectors.Data.GenericASHRAE93 per
+    "Performance data"
+    annotation (choicesAllMatching=true, Placement(transformation(extent={{60,60},{80,80}})));
+
+  Modelica.Blocks.Interfaces.RealInput TIn(
+    final unit = "K",
+    final displayUnit = "degC",
+    quantity = "ThermodynamicTemperature")
+    "Fluid temperature entering the collector"
+    annotation (Placement(transformation(extent={{-140,-60},{-100,-20}})));
+
+  IBPSA.Controls.OBC.CDL.Interfaces.BooleanOutput on
+    "Ouputs true if collector pump should be on"
+    annotation (Placement(transformation(extent={{100,-20},{140,20}})));
+
+  IBPSA.BoundaryConditions.WeatherData.Bus weaBus "Weather data input"
+    annotation (Placement(transformation(extent={{-110,40},{-90,60}})));
+
+  IBPSA.Fluid.SolarCollectors.Controls.BaseClasses.GCritCalc criSol(
+    final slope=per.slope,
+    final y_intercept=per.y_intercept)
+    "Calculates the critical insolation based on collector design and current weather conditions"
+    annotation (Placement(visible = true, transformation(origin={-2,0},    extent = {{-58, -20}, {-38, 0}}, rotation = 0)));
+  BoundaryConditions.SolarIrradiation.DiffusePerez HDifTilIso(
+    final til=til,
+    final azi=azi,
+    final rho=rho) "Diffuse solar irradiation on a tilted surface"
+    annotation (Placement(transformation(extent={{-60,50},{-40,70}})));
+  BoundaryConditions.SolarIrradiation.DirectTiltedSurface HDirTil(
+    final til=til,
+    final azi=azi)
+      "Direct solar irradiation on a tilted surface"
+    annotation (Placement(transformation(extent={{-60,20},{-40,40}})));
+  IBPSA.Controls.OBC.CDL.Reals.Add HTotTil(
+    u1(final unit="W/m2"),
+    u2(final unit="W/m2"),
+    y(final unit="W/m2"))
+    "Total irradiation on tilted surface"
+    annotation (Placement(transformation(extent={{-20,40},{0,60}})));
+
+  IBPSA.Controls.OBC.CDL.Reals.Hysteresis hys(
```

Comment:
Used to be `Buildings.Controls.OBC.CDL.Reals.Hysteresis`, but this model does not exist in the IBPSA library.
I propose we change to `Modelica.Blocks.Logical.Hysteresis`.